src/lib.rs

```diff
@@ -180,8 +197,21 @@ impl<T> StableRegion<T> {
             // We could choose a different rule here if we wanted to be
             // more conservative with memory (e.g. page size allocations).
             let next_len = (self.local.capacity() + 1).next_power_of_two();
-            let new_local = Vec::with_capacity(std::cmp::max(count, next_len));
+            let capacity = std::cmp::max(count, next_len);
+            let capacity = std::cmp::max(capacity, 1<<12 / std::mem::size_of::<T>());
```

Comment:
Consider using `libc::sysconf(libc::_SC_PAGESIZE)` in place of `1<<12`, and rounding up byte_len to a multiple of page size (not strictly necessary since we're mapping a file but cleaner since the allocation will always be in pages anyway).